SensusService/Protocol.cs

```diff
@@ -1080,7 +1236,17 @@ public void StartWithUserAgreementAsync(string startupMessage, Action callback =
 
                         int consentCodeInt;
                         if (int.TryParse(consentCodeStr, out consentCodeInt) && consentCodeInt == consentCode)
-                            Start();
+                        {
+                            // start protocol or schedule start
+                            if (_startImmediately || (_startTime.Date.Equals(DateTime.Now.Date) && DateTime.Now.TimeOfDay > _startTime.TimeOfDay))
```

Comment:
Can we just compare `DateTime.Now` with `_startTime`? What if the user pauses on the consent screen until the date changes to the next day? Using `Equals` will be false.